### ISSUE TITLE
Implement the Rocket Equation

### ISSUE THREAD

**Issue by TheHappyKoala** [TheHappyKoala/Harmony-of-the-Spheres#168]
Right now, the starship scenarios (well, scenario, since there's only one at present, but there will be many more) are not very realistic since there is no limit on the amount of fuel and the spaceship has and therefore also how far you can travel with it. Furthermore, the acceleration induced by burning the engines of the rocket is constant and not at all dependent on the specifications of the starship. 

For the above reasons, it would be nice to introduce the rocket equation - https://en.wikipedia.org/wiki/Tsiolkovsky_rocket_equation. Doing so would entail writing a generic function (so it could be used for any rocket) that calculates the magnitude of the change in velocity caused by burning Starship's rocket engines. This should include a user interface that indicates how much fuel is left in the "tank". It would also be nice to have a checkbox that let the user disable the rocket equation so that they can fly around with limitless fuel.

**Comment by jvanv62** [comment]
Yes!  This is, in fact, the reason I wanted to work on this particular project.  My impression of other simulators is that they just hand-wave over this problem.  Off the top of my head, I was thinking that the engine function should be able to manage different propellant types and have an adjustable energy that follows physics in terms of momentum, but might break the existing energy bank, ie allow for antimatter engine sized energy outputs. As a follow-on, it would be nice to use a similar thing for handle thrusters for changing orientation.

so the module needs to
- track fuel and energy used
- maybe have an adjustment for engine efficiency
- receive mass of spacecraft
- receive mass of fuel remaining
- provide  thrust vector
- provide info on fuel remaining
- monitor energy input
- and a bunch of other things probably
